torch/fx/passes/reinplace.py

```diff
@@ -110,7 +111,17 @@ def run_node(self, node: Node):
     def propagate(self, *args):
         self.multi_output_view_nodes = {}
         self.node_counter = -1
-        with FakeTensorMode.push() as mode:
+        curr_mode = _get_torch_dispatch_mode()
+
+        def assert_fake(x):
+            assert isinstance(x, FakeTensor)
+
+        if type(curr_mode) == FakeTensorMode:
```

Comment:
I'm not a huge fan of this because there might be more modes wedged in between you and the fake tensor mode. Instead, just create FakeTensorMode with `allow_meta=True`; that will allow it to work reentrantly.

Reply:
It looks like this isn't enough: If you try to construct a FakeTensor from an existing FakeTensor, we won't convert the inner FakeTensor to the meta device, which breaks some FakeTensor invariants (the internal tensor is always expected to be on the meta device): https://github.com/pytorch/pytorch/blob/91521449445077c9ee977b18e2d0f19be4dd1c5b/torch/_subclasses/meta_utils.py#L265

I guess I'll update the meta converter to handle nested FakeTensors properly